File: InterviewForge-Backend/src/main/java/com/interviewforge/backend/auth/service/AuthService.java
package com.interviewforge.backend.auth.service;

import com.interviewforge.backend.auth.dto.request.LoginRequest;
import com.interviewforge.backend.auth.dto.request.RegisterRequest;
import com.interviewforge.backend.auth.dto.response.AuthResponse;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
}

File: interviewforge-frontend/src/services/authService.js
import api from "@/api/axios";
import ENDPOINTS from "@/api/endpoints";

export const login = async (credentials) => {
    const response = await api.post(ENDPOINTS.LOGIN, credentials);
    return response.data;
};

export const register = async (data) => {
    const response = await api.post("/auth/register", data);
    return response.data;
};